From 722e795f16ca84d73b4849d0a5cf09f272996915 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 5 Jul 2018 07:44:29 +0200
Subject: [PATCH 1/1] efi_loader: EFI_SIMPLE_TEXT_INPUT_PROTOCOL.Reset()

Implement the reset service of the EFI_SIMPLE_TEXT_INPUT_PROTOCOL.

This should resolve the error reported by the SCT in
Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestFunction.c:193

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_console.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_console.c b/lib/efi_loader/efi_console.c
index ce66c935ec..1d52753456 100644
--- a/lib/efi_loader/efi_console.c
+++ b/lib/efi_loader/efi_console.c
@@ -381,7 +381,12 @@ static efi_status_t EFIAPI efi_cin_reset(
 			bool extended_verification)
 {
 	EFI_ENTRY("%p, %d", this, extended_verification);
-	return EFI_EXIT(EFI_UNSUPPORTED);
+
+	/* Empty input buffer */
+	while (tstc())
+		getc();
+
+	return EFI_EXIT(EFI_SUCCESS);
 }
 
 /*
-- 
2.18.0

